### ISSUE TITLE
xmrig-proxy variant=1 on monero7 not working

### ISSUE THREAD

**Issue by NicNull** [xmrig/xmrig-proxy#164]
I had some issues sending v7 shares via proxy when some config items were a bit "off"
Firstly my old conf had coin "xmr" set and I added variant = 1 when fork hit today. 
(muscle memory from stellite fork, sorry)

My xmr-stak clients have currency = "monero7" set.
My xmrig 2.5.2 clients have variant -1.

It worked a couple of blocks after v7 fork, then started spitting invalid blocks from xmr-stak clients.
When I restarted xmrig-proxy, then all clients including xmrig sent v6 shares.

Went over all settings, did a new conf via config.xmrig.com and diffed old conf. discovered that "coin" was not included and algo should be used.
Swapped to algo="cryptonight" and variant=-1 in proxy, restarted then xmrig seem ok again.

xmr-stak still sending bad shares. 
minexmr pool support also verified v6 frames coming from my proxy.
Changed config at xmr-stak clients to bypass proxy, then it just works fine with the same "monero7" setting as before.
Something is off with regression it seems.

**Comment by xmrig** [comment]
Please show your proxy config.
Thank you.

**Comment by nutsnax** [comment]
having same issue here... xmr-stak not working but xmrig miners are ... screencap:

![image](https://user-images.githubusercontent.com/35117940/38442140-461caba0-39b5-11e8-8d7f-ef5d1428b927.png)

**Comment by agietl** [comment]
I am having the same issue. I have tried changing the algo to monero7. I have also tried changing the variant from -1 to 1 without success.

{
    "access-log-file": null,
    "algo": "cryptonight",
    "api": {
        "port": 8080,
        "access-token": null,
        "worker-id": null,
        "ipv6": true,
        "restricted": true
    },
    "background": false,
    "bind": [
        "0.0.0.0:4444",
        "[::]:4444"
    ],
    "colors": true,
    "custom-diff": 0,
    "donate-level": 2,
    "log-file": null,
    "mode": "nicehash",
    "pools": [
        {
            "url": "pool.supportxmr.com:5555",
            "user": WALLET_ID",
            "pass": `"x",`
            "variant": -1
        }
    ],
    "retries": 2,
    "retry-pause": 1,
    "reuse-timeout": 0,
    "user-agent": null,
    "verbose": false,
    "watch": true,
    "workers": true
}

**Comment by xmrig** [comment]
@nutsnax xmr-stak should be updated to v2.3+ and coin set to monero7, variant option ignored by xmr-stak.
Proxy don't know what algorithm use miner and bypass it as is, miners should care about algorithm.

**Comment by nutsnax** [comment]
I'm on version 2.4.2

@xmrig One thing that make me wonder is that SOME shares are getting through (per the screencap) whereas others are getting rejected... should I disable nicehash support? Or will that break other things?

**Comment by xmrig** [comment]
nicehash support can't be disabled, it will be break anything.
Also xmr-stak now has some open issues with nicehash.com https://github.com/fireice-uk/xmr-stak/issues/1337, https://github.com/fireice-uk/xmr-stak/issues/1222, https://github.com/fireice-uk/xmr-stak/issues/1298 it may related.

**Comment by NicNull** [comment]
I had same configitems as nutsnax.
With "coin"="xmr" removed and instead added algo="cryptonight" then set variant=-1 the normal xmrig miners started working ok again.
As mentioned they worked fine before restart of the proxy. 
I mined over the monerov7 fork threshold with no problems then after a couple of blocks the xmr-stak miners borked. 
nicehash was also turned off when I bypassed proxy to disconnect so could be valid.
